AnimationStateMachinePlayback: Added Missing Method Bindings

Eoin-ONeill-Yokai opened the pull request:
Added missing bindings to `get_current_play_pos` and `get_current_length`.

This allows the user to query the AnimationNodeStateMachinePlayback's current play position and total length of current state. These methods are currently used in the C++ editor plugin, but can also be useful for querying general playback state information for gameplay purposes. It can also be useful for driving state machine playback via script.

Eoin-ONeill-Yokai committed:
StateMachinePlayback: Added missing bindings to `get_current_play_pos` and `get_current_length`.

This allows the user to query the AnimationNodeStateMachinePlayback's current
play position and total length of current animation state. These methods are currently
used in the editor plugin, but can also be useful for querying general playback state
information.

Added documentation for AnimationNodeStateMachinePlayback's `get_current_play_position`

pouleyKetchoupp reviewed (commented):
Seems useful. It would be nice to update the documentation with the two new methods here:
https://github.com/godotengine/godot/blob/master/doc/classes/AnimationNodeStateMachinePlayback.xml

akien-mga commented on scene/animation/animation_node_state_machine.cpp:
```diff
@@ -495,6 +495,8 @@ void AnimationNodeStateMachinePlayback::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("stop"), &AnimationNodeStateMachinePlayback::stop);
 	ClassDB::bind_method(D_METHOD("is_playing"), &AnimationNodeStateMachinePlayback::is_playing);
 	ClassDB::bind_method(D_METHOD("get_current_node"), &AnimationNodeStateMachinePlayback::get_current_node);
+	ClassDB::bind_method(D_METHOD("get_current_play_pos"), &AnimationNodeStateMachinePlayback::get_current_play_pos);
```
This should be renamed to `get_current_play_position` if it's made public.

akien-mga commented:
Indeed if we add methods to the public API, they need documentation.

Eoin-ONeill-Yokai commented:
@akien-mga Sure, just wanted to propose the changes first before writing documentation.

I'll take the time to do that this weekend.

Eoin-ONeill-Yokai commented:
@akien-mga Sorry for the delay, I've been rather busy.

I've added the documentation and corrected the method name in the bindings for GDScript. I've also rebased onto master.

akien-mga commented:
Could you squash the three commits into one? Then it should be good to merge.

Eoin-ONeill-Yokai commented:
@akien-mga It's all merged down to a single commit now. Thanks :)

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.4.
